algosdk/abi/base_type.py

```diff
@@ -126,7 +126,7 @@ def from_string(s: str) -> "ABIType":
                     tuple_list.append(tt)
                 elif isinstance(tup, list):
                     tts = [ABIType.from_string(t_) for t_ in tup]
-                    tuple_list.append(tts)
+                    tuple_list.extend(tts)
```

Comment:
**Sanity check:** is it intentional for `tuple_list` to be typed as `list[ABIType | list[ABIType]]` or should it instead be a flat `list[ABIType]` as I'm assuming here?

Reply:
I think this branch is never entered.  The `_parse_tuple` method will always return a list of strings so this is a dead branch (`isinstance(tup,list)` will always return false). I addressed this in a child PR, but let me know what you think as well: https://github.com/jdtzmn/py-algorand-sdk/pull/1/files.

I think your assumption is right - it should always have a list of `ABITypes`